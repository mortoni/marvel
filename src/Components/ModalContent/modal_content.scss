
.modal_content {
    min-width: 800px;

    .logo {
        border-radius: .6rem;
        padding: 1rem 0;
        border: 1px solid red;

        img {
            width: 6rem;
        }
    }

    .btn-group-vertical {
        button {
            margin: .3rem 0;
        }
    }
}

@media only screen and (max-width: 600px) {
    .modal_content {
        min-width: inherit;
        margin-top: 2.5rem;

        .logo {
            img {
                width: 8rem;
            }
        }
    }
}
