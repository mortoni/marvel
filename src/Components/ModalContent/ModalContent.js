import React from 'react';
import './modal_content.scss';
import PropTypes from 'prop-types';

const ModalContent = (props) => {
    const { thumbnail, name, description } = props;
    const uri = `${thumbnail.path}/portrait_fantastic.${thumbnail.extension}`;
    return (
        <div className="row modal_content">
            <div className="col-xs-12 col-sm-6">
                <div className="logo">
                    <img
                        src={ uri }
                        alt="Marvel character"
                        className="rounded-circle mx-auto d-block"/>
                </div>

                <div className="btn-group-vertical btn-block">
                    <button
                        type="button"
                        className="btn btn-light btn-lg btn-block">
                        Details
                    </button>

                    <button
                        type="button"
                        className="btn btn-light btn-lg btn-block">
                        Comics
                    </button>

                    <button
                        type="button"
                        className="btn btn-light btn-lg btn-block">
                        Events
                    </button>

                    <button
                        type="button"
                        className="btn btn-light btn-lg btn-block">
                        Series
                    </button>

                    <button
                        type="button"
                        className="btn btn-light btn-lg btn-block">
                        Stories
                    </button>
                </div>

            </div>

            <div className="col-xs-12 col-sm-5 text-center">
                <h4>{ name }</h4>
                <p
                    className="text-justify">
                    { description ? description : 'Has no description'}
                </p>
            </div>
        </div>
    );
}
ModalContent.propTypes = {
    thumbnail: PropTypes.object.isRequired,
    name: PropTypes.string.isRequired,
    description: PropTypes.string.isRequired,
};

export default ModalContent
