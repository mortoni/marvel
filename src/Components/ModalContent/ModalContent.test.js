import React from 'react';
import ModalContent from './ModalContent';
import enzymeAdapterReact16 from 'enzyme-adapter-react-16';
import Enzyme, { render, shallow } from 'enzyme';

Enzyme.configure({ adapter: new enzymeAdapterReact16() });

describe('Testing App component', () => {
    const props = {
        thumbnail: {
            extension: 'jpg',
            path: 'http://i.annihil.us/u/prod/marvel/i/mg/b/40/image_not_available'
        },
        name: 'Wolver-dok',
        description: 'blah...'
    };

    it('renders correctly', () => {
        const wrapper = shallow(<ModalContent {...props}/>);
        expect(wrapper).toBeDefined();
    });

    it('renders as expected', () => {
        const wrapper = shallow(<ModalContent {...props}/>);
        expect(wrapper).toMatchSnapshot();
    });
});
