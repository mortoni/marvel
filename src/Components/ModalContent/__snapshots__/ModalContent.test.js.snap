// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing App component renders as expected 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ModalContent
    description="blah..."
    name="Wolver-dok"
    thumbnail={
      Object {
        "extension": "jpg",
        "path": "http://i.annihil.us/u/prod/marvel/i/mg/b/40/image_not_available",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="col-xs-12 col-sm-6"
        >
          <div
            className="logo"
          >
            <img
              alt="Marvel character"
              className="rounded-circle mx-auto d-block"
              src="http://i.annihil.us/u/prod/marvel/i/mg/b/40/image_not_available/portrait_fantastic.jpg"
            />
          </div>
          <div
            className="btn-group-vertical btn-block"
          >
            <button
              className="btn btn-light btn-lg btn-block"
              type="button"
            >
              Details
            </button>
            <button
              className="btn btn-light btn-lg btn-block"
              type="button"
            >
              Comics
            </button>
            <button
              className="btn btn-light btn-lg btn-block"
              type="button"
            >
              Events
            </button>
            <button
              className="btn btn-light btn-lg btn-block"
              type="button"
            >
              Series
            </button>
            <button
              className="btn btn-light btn-lg btn-block"
              type="button"
            >
              Stories
            </button>
          </div>
        </div>,
        <div
          className="col-xs-12 col-sm-5 text-center"
        >
          <h4>
            Wolver-dok
          </h4>
          <p
            className="text-justify"
          >
            blah...
          </p>
        </div>,
      ],
      "className": "row modal_content",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="logo"
            >
              <img
                alt="Marvel character"
                className="rounded-circle mx-auto d-block"
                src="http://i.annihil.us/u/prod/marvel/i/mg/b/40/image_not_available/portrait_fantastic.jpg"
              />
            </div>,
            <div
              className="btn-group-vertical btn-block"
            >
              <button
                className="btn btn-light btn-lg btn-block"
                type="button"
              >
                Details
              </button>
              <button
                className="btn btn-light btn-lg btn-block"
                type="button"
              >
                Comics
              </button>
              <button
                className="btn btn-light btn-lg btn-block"
                type="button"
              >
                Events
              </button>
              <button
                className="btn btn-light btn-lg btn-block"
                type="button"
              >
                Series
              </button>
              <button
                className="btn btn-light btn-lg btn-block"
                type="button"
              >
                Stories
              </button>
            </div>,
          ],
          "className": "col-xs-12 col-sm-6",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <img
                alt="Marvel character"
                className="rounded-circle mx-auto d-block"
                src="http://i.annihil.us/u/prod/marvel/i/mg/b/40/image_not_available/portrait_fantastic.jpg"
              />,
              "className": "logo",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "alt": "Marvel character",
                "className": "rounded-circle mx-auto d-block",
                "src": "http://i.annihil.us/u/prod/marvel/i/mg/b/40/image_not_available/portrait_fantastic.jpg",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <button
                  className="btn btn-light btn-lg btn-block"
                  type="button"
                >
                  Details
                </button>,
                <button
                  className="btn btn-light btn-lg btn-block"
                  type="button"
                >
                  Comics
                </button>,
                <button
                  className="btn btn-light btn-lg btn-block"
                  type="button"
                >
                  Events
                </button>,
                <button
                  className="btn btn-light btn-lg btn-block"
                  type="button"
                >
                  Series
                </button>,
                <button
                  className="btn btn-light btn-lg btn-block"
                  type="button"
                >
                  Stories
                </button>,
              ],
              "className": "btn-group-vertical btn-block",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Details",
                  "className": "btn btn-light btn-lg btn-block",
                  "type": "button",
                },
                "ref": null,
                "rendered": "Details",
                "type": "button",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Comics",
                  "className": "btn btn-light btn-lg btn-block",
                  "type": "button",
                },
                "ref": null,
                "rendered": "Comics",
                "type": "button",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Events",
                  "className": "btn btn-light btn-lg btn-block",
                  "type": "button",
                },
                "ref": null,
                "rendered": "Events",
                "type": "button",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Series",
                  "className": "btn btn-light btn-lg btn-block",
                  "type": "button",
                },
                "ref": null,
                "rendered": "Series",
                "type": "button",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Stories",
                  "className": "btn btn-light btn-lg btn-block",
                  "type": "button",
                },
                "ref": null,
                "rendered": "Stories",
                "type": "button",
              },
            ],
            "type": "div",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h4>
              Wolver-dok
            </h4>,
            <p
              className="text-justify"
            >
              blah...
            </p>,
          ],
          "className": "col-xs-12 col-sm-5 text-center",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Wolver-dok",
            },
            "ref": null,
            "rendered": "Wolver-dok",
            "type": "h4",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "blah...",
              "className": "text-justify",
            },
            "ref": null,
            "rendered": "blah...",
            "type": "p",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="col-xs-12 col-sm-6"
          >
            <div
              className="logo"
            >
              <img
                alt="Marvel character"
                className="rounded-circle mx-auto d-block"
                src="http://i.annihil.us/u/prod/marvel/i/mg/b/40/image_not_available/portrait_fantastic.jpg"
              />
            </div>
            <div
              className="btn-group-vertical btn-block"
            >
              <button
                className="btn btn-light btn-lg btn-block"
                type="button"
              >
                Details
              </button>
              <button
                className="btn btn-light btn-lg btn-block"
                type="button"
              >
                Comics
              </button>
              <button
                className="btn btn-light btn-lg btn-block"
                type="button"
              >
                Events
              </button>
              <button
                className="btn btn-light btn-lg btn-block"
                type="button"
              >
                Series
              </button>
              <button
                className="btn btn-light btn-lg btn-block"
                type="button"
              >
                Stories
              </button>
            </div>
          </div>,
          <div
            className="col-xs-12 col-sm-5 text-center"
          >
            <h4>
              Wolver-dok
            </h4>
            <p
              className="text-justify"
            >
              blah...
            </p>
          </div>,
        ],
        "className": "row modal_content",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="logo"
              >
                <img
                  alt="Marvel character"
                  className="rounded-circle mx-auto d-block"
                  src="http://i.annihil.us/u/prod/marvel/i/mg/b/40/image_not_available/portrait_fantastic.jpg"
                />
              </div>,
              <div
                className="btn-group-vertical btn-block"
              >
                <button
                  className="btn btn-light btn-lg btn-block"
                  type="button"
                >
                  Details
                </button>
                <button
                  className="btn btn-light btn-lg btn-block"
                  type="button"
                >
                  Comics
                </button>
                <button
                  className="btn btn-light btn-lg btn-block"
                  type="button"
                >
                  Events
                </button>
                <button
                  className="btn btn-light btn-lg btn-block"
                  type="button"
                >
                  Series
                </button>
                <button
                  className="btn btn-light btn-lg btn-block"
                  type="button"
                >
                  Stories
                </button>
              </div>,
            ],
            "className": "col-xs-12 col-sm-6",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <img
                  alt="Marvel character"
                  className="rounded-circle mx-auto d-block"
                  src="http://i.annihil.us/u/prod/marvel/i/mg/b/40/image_not_available/portrait_fantastic.jpg"
                />,
                "className": "logo",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "alt": "Marvel character",
                  "className": "rounded-circle mx-auto d-block",
                  "src": "http://i.annihil.us/u/prod/marvel/i/mg/b/40/image_not_available/portrait_fantastic.jpg",
                },
                "ref": null,
                "rendered": null,
                "type": "img",
              },
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <button
                    className="btn btn-light btn-lg btn-block"
                    type="button"
                  >
                    Details
                  </button>,
                  <button
                    className="btn btn-light btn-lg btn-block"
                    type="button"
                  >
                    Comics
                  </button>,
                  <button
                    className="btn btn-light btn-lg btn-block"
                    type="button"
                  >
                    Events
                  </button>,
                  <button
                    className="btn btn-light btn-lg btn-block"
                    type="button"
                  >
                    Series
                  </button>,
                  <button
                    className="btn btn-light btn-lg btn-block"
                    type="button"
                  >
                    Stories
                  </button>,
                ],
                "className": "btn-group-vertical btn-block",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Details",
                    "className": "btn btn-light btn-lg btn-block",
                    "type": "button",
                  },
                  "ref": null,
                  "rendered": "Details",
                  "type": "button",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Comics",
                    "className": "btn btn-light btn-lg btn-block",
                    "type": "button",
                  },
                  "ref": null,
                  "rendered": "Comics",
                  "type": "button",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Events",
                    "className": "btn btn-light btn-lg btn-block",
                    "type": "button",
                  },
                  "ref": null,
                  "rendered": "Events",
                  "type": "button",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Series",
                    "className": "btn btn-light btn-lg btn-block",
                    "type": "button",
                  },
                  "ref": null,
                  "rendered": "Series",
                  "type": "button",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Stories",
                    "className": "btn btn-light btn-lg btn-block",
                    "type": "button",
                  },
                  "ref": null,
                  "rendered": "Stories",
                  "type": "button",
                },
              ],
              "type": "div",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h4>
                Wolver-dok
              </h4>,
              <p
                className="text-justify"
              >
                blah...
              </p>,
            ],
            "className": "col-xs-12 col-sm-5 text-center",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Wolver-dok",
              },
              "ref": null,
              "rendered": "Wolver-dok",
              "type": "h4",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "blah...",
                "className": "text-justify",
              },
              "ref": null,
              "rendered": "blah...",
              "type": "p",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
